### PR TITLE
Remove core-paging dependency in RLC

### PR DESCRIPTION
**Pull request by kazrael2119**
fixes https://github.com/Azure/autorest.typescript/issues/2780

- remove core-paging depedency in RLC
- remove the maxPageSize option in RLC
- remove the [Paged](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-paging/src/models.ts#L69-L78)interface and directly generate paged model type from definition

### CODE REVIEW

File: packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts [hunk 1]
```diff
@@ -1,17 +1,104 @@
...
+  /**
+   * A method that returns a page of results.
+   */
+  getPage: (
+    pageLink: TLink,
+    maxPageSize?: number,
+  ) => Promise<
+    | {
+        page: TPage;
+        nextPageLink?: TLink;
```
Comment (MaryGao) [inline packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts:31]: remove maxPageSize

File: packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts [hunk 2]
```diff
@@ -1,17 +1,104 @@
...
+ */
+export interface PageSettings {
+  /**
+   * The token that keeps track of where to continue the iterator
+   */
+  continuationToken?: string;
+  /**
+   * The size of the page during paged iteration
+   */
+  maxPageSize?: number;
```
Comment (MaryGao) [inline packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts:83]: remove maxPageSize

File: packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts [hunk 3]
```diff
@@ -1,17 +1,104 @@
...
+}
+
+/**
+ * returns an async iterator that iterates over results. It also has a `byPage`
+ * method that returns pages of items at once.
+ *
+ * @param pagedResult - an object that specifies how to get pages.
+ * @returns a paged async iterator that iterates over results.
+ */
+export declare function getPagedAsyncIterator<
```
Comment (MaryGao) [inline packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts:93]: copy the whole implementation part of getPagedAsyncIterator from core to this file.

File: packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts [hunk 4]
```diff
@@ -1,17 +1,180 @@
...
 } from "@azure-rest/core-client";
 
+/**
+ * returns an async iterator that iterates over results. It also has a `byPage`
+ * method that returns pages of items at once.
+ *
+ * @param pagedResult - an object that specifies how to get pages.
+ * @returns a paged async iterator that iterates over results.
+ */
+
```
Comment (MaryGao) [inline packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts:17]:
```suggestion
```

File: packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts [hunk 5]
```diff
@@ -1,17 +1,180 @@
...
+export function getPagedAsyncIterator<
```
Comment (MaryGao) [inline packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/src/paginateHelper.ts:18]:
```suggestion
function getPagedAsyncIterator<
```

File: packages/typespec-test/test/authoring/generated/typespec-ts/review/authoring.api.md [hunk 6]
```diff
@@ -14,8 +14,7 @@ import { ErrorResponse } from '@azure-rest/core-client';
 import { HttpResponse } from '@azure-rest/core-client';
 import { KeyCredential } from '@azure/core-auth';
 import { OperationState } from '@azure/core-lro';
-import { Paged } from '@azure/core-paging';
-import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { Paged as Paged_2 } from '@azure/core-paging';
```
Comment (MaryGao) [inline packages/typespec-test/test/authoring/generated/typespec-ts/review/authoring.api.md:17]: need to remove this.

File: packages/rlc-common/src/metadata/buildPackageFile.ts [hunk 7]
```diff
@@ -108,14 +108,6 @@ export function updatePackageFile(
     packageInfo = existingFilePathOrContent;
   }
 
-  // no need to update paging dependency for modular libraries
-  if (hasPaging && model.options?.isModularLibrary !== true) {
```
Comment (MaryGao) [inline packages/rlc-common/src/metadata/buildPackageFile.ts:112]: remove hasPaging var?

File: packages/rlc-common/src/metadata/packageJson/azurePackageCommon.ts [hunk 8]
```diff
@@ -52,13 +52,6 @@ export function getAzurePackageDependencies({
     };
   }
 
-  if (hasPaging && isModularLibrary !== true) {
```
Comment (MaryGao) [inline packages/rlc-common/src/metadata/packageJson/azurePackageCommon.ts:55]: remove hasPaging var?

File: packages/typespec-test/test/authoring/generated/typespec-ts/review/authoring.api.md [hunk 9]
```diff
@@ -846,7 +862,10 @@ export interface SupportedLanguageOutput {
 }
 
 // @public
-export type SupportedLanguagesOutput = Paged<SupportedLanguageOutput>;
+export interface SupportedLanguagesOutput {
```
Comment (MaryGao) [inline packages/typespec-test/test/authoring/generated/typespec-ts/review/authoring.api.md:865]: @joheredi @qiaozha There is a change in this pr which I believe it is not a breaking and I think it's better to have, could you help review this?

The changes are:
- Paged interface is removed
- The page body model is changed from `Paged<SupportedLanguageOutput>` to directly model defined in TypeSpec.

**Review by MaryGao — APPROVED** [review]

**Review by qiaozha — APPROVED** [review]

**Pull request merged**